service/control.go

```diff
@@ -36,6 +38,11 @@ type postWorkflowReq struct {
 	IsActive    bool      `json:"isActive"` // default false if absent
 }
 
+type deleteWorkflowReq struct {
+	Name     string `json:"name" binding:"required"`
+	Artifact string `json:"artifact" binding:"required"`
```

Comment:
does delete need artifact too?

Reply:
I think safer.  If the artifact is not jvm jar, would it have name conflict?

Reply:
what is the pk of the table?

Reply:
the go orm gorm.model generates a ID column

```
postgres=# \d+ workflows
                                                                   Table "public.workflows"
    Column    |           Type           | Collation | Nullable |                Default                | Storage  | Compression | Stats target | Description
--------------+--------------------------+-----------+----------+---------------------------------------+----------+-------------+--------------+-------------
 id           | bigint                   |           | not null | nextval('workflows_id_seq'::regclass) | plain    |             |              |
 created_at   | timestamp with time zone |           |          |                                       | plain    |             |              |
 updated_at   | timestamp with time zone |           |          |                                       | plain    |             |              |
 deleted_at   | timestamp with time zone |           |          |                                       | plain    |             |              |
 name         | character varying(256)   |           | not null |                                       | extended |             |              |
 artifact     | character varying(2048)  |           | not null |                                       | extended |             |              |
 command      | text                     |           | not null |                                       | extended |             |              |
 every        | character varying(64)    |           | not null |                                       | extended |             |              |
 next_runtime | timestamp with time zone |           | not null |                                       | plain    |             |              |
 backfill     | boolean                  |           | not null |                                       | plain    |             |              |
 owner        | character varying(2048)  |           |          |                                       | extended |             |              |
 is_active    | boolean                  |           | not null |                                       | plain    |             |              |
Indexes:
    "workflows_pkey" PRIMARY KEY, btree (id)
    "idx_workflows_deleted_at" btree (deleted_at)
    "workflows_name_artifact" UNIQUE, btree (name, artifact)
Referenced by:
    TABLE "scheduled_workflows" CONSTRAINT "fk_workflows_scheduled_workflows" FOREIGN KEY (workflow_id) REFERENCES workflows(id)
Access method: heap
```

Reply:
okay, looks the same as starport. I think can just delete by name.

Reply:
okay.  new commit pushed